homeassistant/components/geocaching/config_flow.py

```diff
@@ -0,0 +1,65 @@
+"""Config flow for Geocaching."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from geocachingapi.geocachingapi import GeocachingApi
+from geocachingapi.models import GeocachingUser
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.config_entry_oauth2_flow import AbstractOAuth2FlowHandler
+
+from .const import DOMAIN
+
+
+class GeocachingFlowHandler(AbstractOAuth2FlowHandler, domain=DOMAIN):
+    """Config flow to handle Geocaching OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+    # Geocaching is polling from the cloud
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+    VERSION = 1
+    data: Any = {}
```

Comment:
Don't initialize a dictionary in a class variable, as this will apply/be the same dict for all instances of this class.

And actually, you won't need it.

```suggestion
```

Reply:
True!